shogun-lib/src/main/java/de/terrestris/shogun/lib/service/BaseService.java

```diff
@@ -131,4 +138,49 @@ public void delete(S entity) {
 
         repository.delete(entity);
     }
+
+    /**
+     * Get a historic {@link BaseEntity} for a given time
+     *
+     * @param id The id of the entity
+     * @param time
+     * @return A single {@link BaseEntity}
+     */
+    @PostAuthorize("hasRole('ROLE_ADMIN') or hasPermission(returnObject.orElse(null), 'READ')")
+    @Transactional(readOnly = true)
+    public Optional<S> findOneByTime(
+        Long id,
+        OffsetDateTime time
+    ) {
+        Class<? extends BaseEntity> clazz = getBaseEntityClass();
+        if (clazz == null) {
+            return Optional.empty();
+        }
+        List<S> revisions = (List<S>) auditReader.createQuery()
```

Comment:
Do you know if the auditReader is taking advantage of the hibernate caches as well?